feat(docs): separate Airbyte connectors and community connectors in sidebar

letiescanciano opened the pull request:
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/8332
## What
This PR creates new items in the docs sidebar to better organize connectors  depending on its support level.


![Screenshot 2025-05-07 at 08 16 04](https://github.com/user-attachments/assets/c79ccaa8-144f-42a8-8b6b-e7281555fb27)


## How
- Fetching registry information on pre-build to make sure is available for the sidebar
- Finding support level information on the registry
- Grouping connectors by support level
- Displaying in the sidebar


## Can this PR be safely reverted and rolled back?
<!--
* If unsure, leave it blank.
-->
- [x] YES 💚
- [ ] NO ❌

letiescanciano committed:
feat:docs separate airbyte connectors and community connectors in docs sidebar

letiescanciano commented:
@bnchrch @ian-at-airbyte  in case you have a minute to review this :)

ian-at-airbyte commented on docusaurus/src/theme/DocSidebar/utils.js:
```diff
@@ -0,0 +1,48 @@
+        collapsible: true,
+        collapsed: true,
+        label:
+          supportLevel === "certified" ? "Airbyte Connectors" : "Marketplace",
```
Could we standardize how we label these? IE we should either use:

"Airbyte" + "Marketplace"
or
"Airbyte Connectors" and "Marketplace Connectors".

ian-at-airbyte reviewed (commented):
I like this idea, but I am a bit cold on how it's being reflected on the site itself. I'll explain what I mean.

The site architecture implied by the Sidebar is not implied by the breadcrumbs. In this case, the breadcrumbs skip a level between Sources and the connector.

<img width="1286" alt="image" src="https://github.com/user-attachments/assets/5aa2344b-35ab-4945-ab22-a4aa0ca5c01b" />

The same thing exists with the search results - for example, if you search for Airbtable, its parent section is listed as "Sources, not "Airbyte Connectors" as I would expect.

_I think_ what's happening is that the true site structure is defined by Sidebars.js. Everything else inherits from this. sidebars.js does not have the information about the DocSidebar component to know that this new logic should form part of the site structure. So the sidebar gets updated but nothing that depends on it does.

How possible is it that, instead of swizzling DocSidebar, the functions in sidebars.js can do this work instead? I think that would inform the breadcrumbs and Algolia that they should behave this way as well.

letiescanciano commented:
@ian-at-airbyte thanks for your comment! it makes a lot of sense. I dig deeper on how the breadcrums and algolia search are linked to the sidebar.

letiescanciano committed:
Merge branch 'master' into leti/separate-airbyte-and-marketplace-connectors-in-sidebar

* master: (3385 commits)
  fix(Source-Harvest): missing schema for `time_team`, `users`, `time_entries`, `company`, & `project_assignments` streams.  (#59215)
  chore: add `poe_tasks.toml` for all java connectors (#59670)
  ci: add Dockerfile base image definitions, along with sample `.sh` build scripts (#59212)
  Correct Dest SF readme. (#59658)
  ✨ TikTok Marketing Source: By country daily/hourly report streams (#55681)
  source-vitally contribution from aidanlister (#58062)
  Upgrade Gradle to 8.14. (#59662)
  source-awin-advertiser contribution from ryanmcg2203 (#58632)
  🐛 Source asana: add new stream portfolio_items (#59224)
  docs: add pre 1.0 semver exception to breaking changes
  ✨ Source Brevo: fix pagination on contact stream (#59652)
  Load CDK: Pre-Speed Cleanup: Remove CheckpointManager interface/abstract class (#59218)
  Load CDK: Pre-Speed Cleanup: Remove Sync/Stream manager interfaces (#59217)
  Load CDK: Non-functional Cleanup: Remove Launcher Interface/Secondary (#59213)
  fix(live-tests): use first 8 symbols of connection id instead of hashing (#59653)
  ✨feat(source-s3): adjust file record message protocol (#57498)
  feat(source-zendesk-support): new article attachments with file transfer support  (#56419)
  source-mixmax contribution from cbuckle1 (#59160)
  ✨feat(source-google-drive): adjust file record message protocol (#57496)
  Typo fix (#59209)
  ...

letiescanciano commented:
> I like this idea, but I am a bit cold on how it's being reflected on the site itself. I'll explain what I mean.
> 
> The site architecture implied by the Sidebar is not implied by the breadcrumbs. In this case, the breadcrumbs skip a level between Sources and the connector.
> 
> <img alt="image" width="1286" src="https://private-user-images.githubusercontent.com/187576150/429168979-5aa2344b-35ab-4945-ab22-a4aa0ca5c01b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY1MDg1MzUsIm5iZiI6MTc0NjUwODIzNSwicGF0aCI6Ii8xODc1NzYxNTAvNDI5MTY4OTc5LTVhYTIzNDRiLTM1YWItNDk0NS1hYjIyLWE0YWEwY2E1YzAxYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUwNlQwNTEwMzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NjliZTQ0ZDEyNjk0YmEzNTAxODNkZDM3NTBmMDFiZTVhMTEzZTVhOWQwODgwYjVkNDg0YmRkNTBjNGYwOWQzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.sUkx5-TkanKm9o5TP6YSPL3IyUOQsW6g2zzLrP-aunI">
> The same thing exists with the search results - for example, if you search for Airbtable, its parent section is listed as "Sources, not "Airbyte Connectors" as I would expect.
> 
> _I think_ what's happening is that the true site structure is defined by Sidebars.js. Everything else inherits from this. sidebars.js does not have the information about the DocSidebar component to know that this new logic should form part of the site structure. So the sidebar gets updated but nothing that depends on it does.
> 
> How possible is it that, instead of swizzling DocSidebar, the functions in sidebars.js can do this work instead? I think that would inform the breadcrumbs and Algolia that they should behave this way as well.

The reason to swizzling DocSidebar was that, in order to divide connectors by support level, I needed to fetch registry information and that couldn't be done directly in sidebar.js. I'll try to create a plugin to handle it

letiescanciano committed:
feat: implement support level-based connector categorization

- Add support level categorization for connectors (Airbyte, Marketplace, Enterprise)
- Add prepare-sidebar script to generate connector registry data
- Modify sidebar structure to group connectors by support level
- Update special connector handling to include support level information
- Remove separate enterprise connectors section in favor of support level grouping
- Update support level display names for consistency

This change improves the connector documentation organization by grouping
connectors based on their support level rather than type, making it easier
for users to find connectors based on their support tier.

letiescanciano committed:
fix format

letiescanciano commented:
@ian-at-airbyte I was able to build the sidebar with the new structure without swizzling DocSidebar.
Now, the breadcrumbs are following the same structure
Let me know what you think!

letiescanciano commented:
>Would we have to manually generate the connector registry in this case?
no, it will fetch the registry at build time and create a slim version of it, which the sidebar be build on.

letiescanciano committed:
collapse support level items

ian-at-airbyte reviewed (approved):
Brilliant.
